examples/seq2seq/run_seq2seq.py

```diff
@@ -212,6 +219,31 @@ def __post_init__(self):
 }
 
 
+def freeze_params(model: nn.Module):
+    """Set requires_grad=False for each of model.parameters()"""
+    for par in model.parameters():
+        par.requires_grad = False
+
+
+def freeze_embeds(model):
+    """Freeze token embeddings and positional embeddings for bart, just token embeddings for t5."""
+    model_type = model.config.model_type
+
+    if model_type == "t5":
+        freeze_params(model.shared)
+        for d in [model.encoder, model.decoder]:
+            freeze_params(d.embed_tokens)
+    elif model_type == "fsmt":
+        for d in [model.model.encoder, model.model.decoder]:
+            freeze_params(d.embed_positions)
+            freeze_params(d.embed_tokens)
```

Comment:
Think this is not super important and can remove it.  We will need to add `get_pos_embeddings` and `get_token_embeddings` methods in every model to avoid handling such special cases here. AFAIK this was added to save some memory and speed-up training, but now with `fairseq/deepspeed` integration, we have better ways of doing that.

Reply:
Not super fond of this, so yeah maybe remove the `freeze_embeds` option for this script?